bip78/src/receiver/mod.rs

```diff
@@ -113,3 +163,62 @@ pub struct NewOutputOptions {
     subtract_fees_from_this: bool,
 }
 
+mod test {
+    use super::*;
+
+    struct MockHeaders {
+        length: String,
+    }
+
+    impl MockHeaders {
+        #[cfg(test)]
+        fn new(length: u64) -> MockHeaders {
+            MockHeaders { length: length.to_string() }
+        }
+    }
+
+    impl Headers for MockHeaders {
+        fn get_header(&self, key: &str) -> Option<&str> {
+            match key {
+                "content-length" => Some(&self.length),
+                "content-type" => Some("text/plain"),
+                _ => None,
+            }
+        }
+    }
```

Comment:
tests/integration.rs also contains HeaderMock. Perhaps they belong in some common testutils.